inc/class-wp-sitemaps-renderer.php

```diff
@@ -255,11 +255,11 @@ static function () {
 
 			wp_die(
 				sprintf(
-				/* translators: %s: SimpleXML */
-					esc_xml( __( 'Could not generate XML sitemap due to missing %s extension', 'core-sitemaps' ) ),
+					/* translators: %s: SimpleXML */
+					__( 'Could not generate XML sitemap due to missing %s extension', 'core-sitemaps' ),
 					'SimpleXML'
 				),
-				esc_xml( __( 'WordPress &rsaquo; Error', 'core-sitemaps' ) ),
```

Comment:
This one does need to call `esc_xml()`, so that `&rsaquo;` gets escaped to `›`.  The other one _should_ be XML escaped, in case there are HTML entities in a translation.

FYI: when the core merge happens, `_xml_wp_die_handler()` should be rewritten to call `esc_xml()`, instead of `htmlspecialchars()` on both the `$message` and `$title` params...and to **not** wrap those in CDATA sections in the output; and when happens these can both be changed back to `__()`.

Reply:
Just so I get this right, the `esc_xml` is still needed, despite the CDATA sections in `_xml_wp_die_handler`?

Reply:
yes...so that HTML entities get converted to their code points.

Reply:
Got it, thanks! Guess I was a bit over-excited about removing potentially unneeded code. But it all makes sense now :-)

Reply:
BTW, the `$learn_more` link is a much cleaner way to handle that in the stylesheets!

You educate me about l10n and I educate you about XML :-)